javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -1,31 +1,1669 @@
+
+      val startIndex = 0
+      val flags      = formatToken.getFlags()
+
+      if (formatToken.isFlagSet(FormatToken.FLAG_MINUS) && !formatToken
+            .isWidthSet())
+        throw new MissingFormatWidthException(
+          "-" + formatToken.getConversionType())
+
+      if (FormatToken.FLAGS_UNSET != flags && FormatToken.FLAG_MINUS != flags)
+        throw new FormatFlagsConversionMismatchException(
+          formatToken.getStrFlags(),
+          formatToken.getConversionType())
+
+      if (formatToken.isPrecisionSet())
+        throw new IllegalFormatPrecisionException(formatToken.getPrecision())
+
+      arg.asInstanceOf[Any] match {
+        case null         => result.append("null")
+        case c: Character => result.append(c)
+        case b: Byte =>
+          if (!Character.isValidCodePoint(b))
+            throw new IllegalFormatCodePointException(b)
+          result.append(b.asInstanceOf[Char])
+        case s: Short =>
+          if (!Character.isValidCodePoint(s))
+            throw new IllegalFormatCodePointException(s)
+          result.append(s.asInstanceOf[Char])
+        case codePoint: Int =>
+          if (!Character.isValidCodePoint(codePoint))
+            throw new IllegalFormatCodePointException(codePoint)
+          result.append(String.valueOf(Character.toChars(codePoint)))
+        case _ =>
+          throw new IllegalFormatConversionException(
+            formatToken.getConversionType(),
+            arg.getClass())
+      }
+      padding(result, startIndex)
+    }
+
+    private def transformFromPercent(): String = {
+      val result = new StringBuilder("%")
+
+      val startIndex = 0
+      val flags      = formatToken.getFlags()
+
+      if (formatToken.isFlagSet(FormatToken.FLAG_MINUS) && !formatToken
+            .isWidthSet())
+        throw new MissingFormatWidthException(
+          "-" + formatToken.getConversionType())
+
+      if (FormatToken.FLAGS_UNSET != flags && FormatToken.FLAG_MINUS != flags)
+        throw new FormatFlagsConversionMismatchException(
+          formatToken.getStrFlags(),
+          formatToken.getConversionType())
+      if (formatToken.isPrecisionSet())
+        throw new IllegalFormatPrecisionException(formatToken.getPrecision())
+      padding(result, startIndex)
+    }
+
+    private def transformFromLineSeparator(): String = {
+      if (formatToken.isPrecisionSet())
+        throw new IllegalFormatPrecisionException(formatToken.getPrecision())
+
+      if (formatToken.isWidthSet())
+        throw new IllegalFormatWidthException(formatToken.getWidth())
+
+      val flags = formatToken.getFlags()
+      if (FormatToken.FLAGS_UNSET != flags)
+        throw new IllegalFormatFlagsException(formatToken.getStrFlags())
+
+      if (null == lineSeparator) {
+        lineSeparator =
+          AccessController.doPrivileged(new PrivilegedAction[String]() {
+            def run(): String = System.getProperty("line.separator")
+          })
```

Comment:
This is going to become just `System.getProperty("line.separator")` without access control wrapper.